### PR TITLE
Document `AbstractRNG` and `Random.Random`

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

File: stdlib/Random/src/Random.jl [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
+"""
```
Comment (fredrikekre) [inline stdlib/Random/src/Random.jl]:
```suggestion
"""
    Random

```

File: stdlib/Random/src/Random.jl [hunk 2]
```diff
@@ -27,6 +31,11 @@ export rand!, randn!,
 
 ## general definitions
 
+"""
+    AbstractRNG
+
+Supertype for random number generators. [`MersenneTwister`](@ref) and [`RandomDevice`](@ref) are subtypes of this.
```
Comment (fredrikekre) [inline stdlib/Random/src/Random.jl]:
```suggestion
Supertype for random number generators such as [`MersenneTwister`](@ref) and [`RandomDevice`](@ref).
```

**Comment by kshyatt** [comment]
Odd error in `factorial` `OverFlowError` test...

**Comment by fredrikekre** [comment]
#31288

**Pull request merged**